opencensus-shim/src/main/java/io/opentelemetry/opencensusshim/OpenTelemetrySpanImpl.java

```diff
@@ -0,0 +1,238 @@
+  }
+
+  @Override
+  public void putAttribute(String key, AttributeValue value) {
+    Preconditions.checkNotNull(key, "key");
+    Preconditions.checkNotNull(value, "value");
+    value.match(
+        arg -> otSpan.setAttribute(key, arg),
+        arg -> otSpan.setAttribute(key, arg),
+        arg -> otSpan.setAttribute(key, arg),
+        arg -> otSpan.setAttribute(key, arg),
+        arg -> null);
+  }
+
+  @Override
+  public void putAttributes(Map<String, AttributeValue> attributes) {
+    Preconditions.checkNotNull(attributes, "attributes");
+    for (Map.Entry<String, AttributeValue> attribute : attributes.entrySet()) {
+      putAttribute(attribute.getKey(), attribute.getValue());
+    }
+  }
+
+  @Override
+  public void addAnnotation(String description, Map<String, AttributeValue> attributes) {
+    AttributesBuilder attributesBuilder = Attributes.builder();
+    attributes.forEach(
+        (s, attributeValue) ->
+            attributeValue.match(
+                setStringAttribute(attributesBuilder, s),
+                setBooleanAttribute(attributesBuilder, s),
+                setLongAttribute(attributesBuilder, s),
+                setDoubleAttribute(attributesBuilder, s),
+                arg -> null));
+    otSpan.addEvent(description, attributesBuilder.build());
+  }
+
+  @Override
+  public void addAnnotation(Annotation annotation) {
+    AttributesBuilder attributesBuilder = Attributes.builder();
+    annotation
+        .getAttributes()
+        .forEach(
+            (s, attributeValue) ->
+                attributeValue.match(
+                    setStringAttribute(attributesBuilder, s),
+                    setBooleanAttribute(attributesBuilder, s),
+                    setLongAttribute(attributesBuilder, s),
+                    setDoubleAttribute(attributesBuilder, s),
+                    arg -> null));
+    otSpan.addEvent(annotation.getDescription(), attributesBuilder.build());
+  }
+
+  @Override
+  public void addLink(Link link) {
+    LOGGER.warning("OpenTelemetry does not support links added after a span is created.");
+  }
+
+  @Override
+  public void addMessageEvent(MessageEvent messageEvent) {
+    otSpan.addEvent(
+        String.valueOf(messageEvent.getMessageId()),
+        Attributes.of(
+            AttributeKey.stringKey(MESSAGE_EVENT_ATTRIBUTE_KEY_TYPE),
+            messageEvent.getType().toString(),
+            AttributeKey.longKey(MESSAGE_EVENT_ATTRIBUTE_KEY_SIZE_UNCOMPRESSED),
+            messageEvent.getUncompressedMessageSize(),
+            AttributeKey.longKey(MESSAGE_EVENT_ATTRIBUTE_KEY_SIZE_COMPRESSED),
+            messageEvent.getCompressedMessageSize()));
+  }
+
+  @Override
+  public void setStatus(Status status) {
+    Preconditions.checkNotNull(status, "status");
+    otSpan.setStatus(status.isOk() ? StatusCode.OK : StatusCode.ERROR);
+  }
+
+  @Override
+  public io.opentelemetry.api.trace.Span setStatus(StatusCode canonicalCode, String description) {
+    return null;
```

Comment:
```suggestion
    otSpan.setStatus(canonicalCode, description);
    return this;
```

Don't we need to delegate to the OTel span for all of these methods too since they might be called from OTel instrumentation?

Also make sure to `return this` everywhere

Reply:
I think only OpenCensus instrumentation will interact with these spans, unless for some reason a user decided to get an OC parent/child span from a library and change its status. I've updated the Otel methods to delegate to the otelSpan just in case though.